antispam_bee.php

```diff
@@ -2348,32 +2350,50 @@ public static function get_client_ip() {
 			$ip = wp_unslash( $_SERVER['HTTP_FORWARDED_FOR'] );
 		} elseif ( isset( $_SERVER['HTTP_FORWARDED'] ) ) {
 			$ip = wp_unslash( $_SERVER['HTTP_FORWARDED'] );
-		} elseif ( isset( $_SERVER['REMOTE_ADDR'] ) ) {
-			$ip = wp_unslash( $_SERVER['REMOTE_ADDR'] );
-		} else {
-			return '';
 		}
-		// phpcs:enable WordPress.VIP.ValidatedSanitizedInput.InputNotSanitized
 
-		if ( strpos( $ip, ',' ) !== false ) {
-			$ips = explode( ',', $ip );
-			$ip  = trim( $ips[0] );
+		$ip = self::_sanitize_ip( $ip );
+		if ( $ip ) {
+			return $ip;
+		}
+
+		if ( isset( $_SERVER['REMOTE_ADDR'] ) ) {
+			$ip = wp_unslash( $_SERVER['REMOTE_ADDR'] );
+			return self::_sanitize_ip( $ip );
 		}
 
+		return '';
+        // phpcs:enable WordPress.VIP.ValidatedSanitizedInput.InputNotSanitized
+	}
+
+	/**
+	 * Sanitize an IP string.
+	 *
+	 * @param string $raw_ip The raw IP.
+	 *
+	 * @return string The sanitized IP or an empty string.
+	 */
+	private static function _sanitize_ip( $raw_ip ) {
+
+		if ( strpos( $raw_ip, ',' ) !== false ) {
+			$ips    = explode( ',', $raw_ip );
+			$raw_ip = trim( $ips[0] );
+		}
 		if ( function_exists( 'filter_var' ) ) {
-			return filter_var(
-				$ip,
+			return (string) filter_var(
+				$raw_ip,
 				FILTER_VALIDATE_IP
 			);
 		}
 
-		return preg_replace(
+		return (string) preg_replace(
 			'/[^0-9a-f:\., ]/si',
```

Comment:
Comma should no longer occur at this point and escaping the dot within the character class is redundant.

Reply:
good catch